lens-cube/src/main/java/org/apache/lens/cube/parse/CubeQueryRewriter.java

```diff
@@ -150,13 +150,14 @@ private void setupRewriters() {
     // Resolve candidate fact tables and dimension tables for columns queried
     rewriters.add(candidateTblResolver);
     // Resolve aggregations and generate base select tree
+    rewriters.add(new AggregateResolver());
+    rewriters.add(new GroupbyResolver(conf));
+    rewriters.add(new FieldValidator());
+    rewriters.add(storageTableResolver);
     rewriters.add(new CandidateCoveringSetsResolver(conf));
 
     //TODO union: Add CoveringSetResolver which creates UnionCandidates and JoinCandidates. Some code form candidateTblResolver(phase 2) to be moved to CoveringSetResolver
```

Comment:
I think we can keep this for our reference, when we will run test cases.  Plenty of test cases will fail we can refer our discussion here. May be we can clean these during final cleanup.